smsdk/utils.py

```diff
@@ -26,4 +40,40 @@ def get_url(protocol, tenant, site_domain):
     :type site_domain: :class:`string`
     """
 
-    return "{}://{}.{}".format(protocol, tenant, site_domain)
\ No newline at end of file
+    return "{}://{}.{}".format(protocol, tenant, site_domain)
+
+def check_kw(kw: str) -> bool:
+    for key in [
+        "machine_type", 
+        "Machine",
+        "machine__source", 
+        "End Time",
+        "endtime",
+        "Start Time",
+        "starttime"
+    ]:
+        if kw.startswith("_") or key in kw:
+            return False
+    return True
```

Comment:
Need a comment on the purpose of this function also do not need to check `kw` repeatedly.  Do we want to be case insensitive?
```suggestion
def check_kw(kw: str) -> bool:
    return not (
        kw.startswith("_")
        or any(
            i in kw
            for i in (
                "machine_type", 
                "Machine",
                "machine__source", 
                 "End Time",
                "endtime",
                "Start Time",
                "starttime"
            )
        )
    )

Reply:
No, we don't want it to be case sensitive as it is there to support the v0 style of passing arguments to sdk methods, if we stop supporting this style, it may break some applications which are being used with sdk, for that reason I've retained this style of coding.

Reply:
Also not included the suggestion as I found the suggested code is performing more or less the same in most cases, here is the result for that

**old code**
CPU times: user 11 µs, sys: 0 ns, total: 11 µs
Wall time: 13.8 µs

**suggested code**
CPU times: user 13 µs, sys: 0 ns, total: 13 µs
Wall time: 15.7 µs